#!/usr/bin/env python
# HANs AGV configuration

PACKAGE = "agv_odom"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_vel", double_t, 0, "Linear velocity.", 0.7, -3.0, 3.0)

gen.add("angular_vel", double_t, 0, "Angular speed in radians per second", 1.0, -2.5, 2.5)

gen.add("left_motor_speed", int_t, 0, "Left wheel rolling speed in RMP", 100, -3000, 3000)

gen.add("right_motor_speed", int_t, 0, "Angular correction factor", 100, -3000, 3000)

gen.add("start", bool_t, 0, "Check to move the mobile base", False)
gen.add("stop", bool_t, 0, "Check to stop the mobile base", False)

gen.add("kp", double_t, 0, "Proportional Const", 0, -1000.0, 1000.0)
gen.add("ki", double_t, 0, "Integral Const", 0, -1000.0, 1000.0)
gen.add("kd", double_t, 0, "Derivative Const", 0, -1000.0, 1000.0)

gen.add("restore_defaults", bool_t, 0, "Restore to the original configuration.", False)

exit(gen.generate(PACKAGE, "boost_socket_test_node", "HANsAGVMotion"))